eZ/Publish/Core/Persistence/Legacy/Content/Handler.php

```diff
@@ -554,6 +554,11 @@ public function deleteVersion( $contentId, $versionNo )
         $this->contentGateway->deleteRelations( $contentId, $versionNo );
         $this->contentGateway->deleteVersions( $contentId, $versionNo );
         $this->contentGateway->deleteNames( $contentId, $versionNo );
+
+        if ( $this->contentGateway->getLastVersionNumber( $contentId ) == 0 )
```

Comment:
this check is a bit annoying to me... i mean, when a content version number can be 0? i've seen that this may come from https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php#L1085, but imho we shouldn't return 0 when there is no version. i will vote for null instead... 
it's a bit weird that 0 could be the last number of a row :).